### PR TITLE
Mahitha | BAH-477 | Save patient even entered patient Address does not e…

### PR DESCRIPTION
**Pull request by Mahitha1467**
…xist in database

### CODE REVIEW

**Review by mks-d — COMMENTED** [review]
@Mahitha1467 please write the full JIRA code (`BAH-477`, not `#477`) in the headline of your commit (for JIRA to sniff it).

File: omod/src/main/java/org/bahmni/module/bahmniOfflineSync/utils/AddressCalculator.java [hunk 1]
```diff
@@ -51,7 +51,11 @@ public void addTopDownAddressFor(Patient patient) {
             AddressHierarchyLevel addressHierarchyLevel = addressHierarchyService.getTopAddressHierarchyLevel();
             String addressString = getAddressValueForLevel(addressHierarchyLevel.getLevelId(), address);
             List<AddressHierarchyEntry> childAddressHierarchyEntry = addressHierarchyService.getAddressHierarchyEntriesByLevelAndName(addressHierarchyLevel, addressString);
-            addressCode = getAddressCode(addressHierarchyLevel.getLevelId(), childAddressHierarchyEntry.get(0), address);
+            if(childAddressHierarchyEntry == null || childAddressHierarchyEntry.isEmpty()) {
```
Comment (sumanmaity112) [inline omod/src/main/java/org/bahmni/module/bahmniOfflineSync/utils/AddressCalculator.java]: @Mahitha1467,

Can you please move this `if else` block (line no 54 to 58) to a method

File: omod/src/test/java/org/bahmni/module/bahmniOfflineSync/utils/AddressCalculatorIT.java [hunk 2]
```diff
@@ -44,6 +44,25 @@ public void shouldAddAddressCodeCalculatedTopDownForCity() throws Exception {
         Assert.assertEquals(patient.getAttribute("addressCode").getValue(), "35");
     }
 
+    @Test
+    public void shouldNotHaveAddressCodeAttributeIfThePatientDoesNotHaveAddress() {
+        AddressCalculator addressCalculator = new AddressCalculator();
+        Patient patient = new Patient();
+
+        addressCalculator.addTopDownAddressFor(patient);
+        Assert.assertEquals(null, patient.getAttribute("addressCode"));
```
Comment (sumanmaity112) [inline omod/src/test/java/org/bahmni/module/bahmniOfflineSync/utils/AddressCalculatorIT.java]: @Mahitha1467,
Please use `assertNull` instead `assertEquals`

File: omod/src/test/java/org/bahmni/module/bahmniOfflineSync/utils/AddressCalculatorIT.java [hunk 3]
```diff
@@ -44,6 +44,25 @@ public void shouldAddAddressCodeCalculatedTopDownForCity() throws Exception {
...
+    }
+
+    @Test
+    public void shouldNotHaveAddressCodeAttributeIfTheGivenCountryIsNotExistInDatabase() {
+        AddressCalculator addressCalculator = new AddressCalculator();
+        Patient patient = buildPatientWithAddress("randomText", "Telangana", "Secunderabad", "Gachibowli");
+
+        addressCalculator.addTopDownAddressFor(patient);
+        Assert.assertEquals(null, patient.getAttribute("addressCode"));
```
Comment (sumanmaity112) [inline omod/src/test/java/org/bahmni/module/bahmniOfflineSync/utils/AddressCalculatorIT.java]: Please use `assertNull` instead `assertEquals`

File: omod/src/test/java/org/bahmni/module/bahmniOfflineSync/utils/AddressCalculatorIT.java [hunk 4]
```diff
@@ -44,6 +44,25 @@ public void shouldAddAddressCodeCalculatedTopDownForCity() throws Exception {
...
+    }
+
+
```
Comment (sumanmaity112) [inline omod/src/test/java/org/bahmni/module/bahmniOfflineSync/utils/AddressCalculatorIT.java]: Please remove this empty line. It should have max 1 empty line

**Review by krishnanspace — COMMENTED** [review]
Everything seems fine to me.  
@binduak any comments?

**Comment by sumanmaity112** [comment]
@krishnanspace, please don't merge it now. Please have a look into [this](https://talk.openmrs.org/t/bahmni-connect-getting-error-while-saving-patient-with-address-which-doesnt-exist-in-database/16205/9) comment.

**Comment by krishnanspace** [comment]
@sumanmaity112 Not merging.Was just reviewing the code

**Pull request closed without merging**